plugins/events-backend/src/schema/openapi.yaml

```diff
@@ -0,0 +1,182 @@
+    JWT:
+      type: http
+      scheme: bearer
+      bearerFormat: JWT
+paths:
+  /bus/v1/events:
+    post:
+      operationId: PostEvent
+      description: Publish a new event
+      responses:
+        '201':
+          description: The event was published successfully
+        '204':
+          description: The event did not need to be published as all subscribers have already been notified
+        default:
+          $ref: '#/components/responses/ErrorResponse'
+      security:
+        - {}
+        - JWT: []
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              required:
+                - event
+              properties:
+                event:
+                  $ref: '#/components/schemas/Event'
+                notifiedSubscribers:
+                  type: array
+                  description: The IDs of subscriptions that have already received this event
+                  items:
+                    type: string
+            examples:
+              Publishing a simple Event:
+                value:
+                  event:
+                    topic: test-topic
+                    payload:
+                      myData: foo
+
+  /bus/v1/subscriptions/{subscriptionId}:
+    put:
+      operationId: PutSubscription
+      description: Ensures that the subscription exists with the provided configuration
+      responses:
+        '201':
+          description: The subscription exists or was created successfully
+        default:
+          $ref: '#/components/responses/ErrorResponse'
+      security:
+        - {}
+        - JWT: []
+      parameters:
+        - $ref: '#/components/parameters/subscriptionId'
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              required:
+                - topics
+              properties:
+                topics:
+                  type: array
+                  description: The topics to subscribe to
+                  items:
+                    type: string
+            examples:
+              Subscribing to a single topic:
+                value:
+                  topics:
+                    - test-topic
+
+  /bus/v1/subscriptions/{subscriptionId}/events:
+    get:
+      operationId: GetSubscriptionEvents
+      description: Get new events for the provided subscription
+      responses:
+        '200':
+          description: New events
+          content:
+            application/json:
+              schema:
+                type: object
+                required:
+                  - events
+                properties:
+                  events:
+                    type: array
+                    items:
+                      $ref: '#/components/schemas/Event'
+        '202':
+          description: No new events are available. Response will block until the client should try again.
```

Comment:
Can't really retry-after since we don't know when that would be. The purpose of the blocking response is to let the caller know immediately when an event is available, rather than frequent polling. I think the busy loop is something to protect against in the client implementation for this kind of thing, not in the API. Definitely gotta make sure errors are handled well on the client-side ofc

Reply:
Yeah of course it would introduce delays to do polling.